deploy/helm/sumologic/templates/_helpers.tpl

```diff
@@ -1510,111 +1526,87 @@ Example Usage:
 {{- printf "%s.%s" ( include "sumologic.metadata.name.otelcol.service-headless" . ) .Release.Namespace }}
 {{- end -}}
 
-{{- define "opentelemetry-operator.webhook.service.name" -}}
-opentelemetry-operator-webhook-service
-{{- end -}}
-
-{{- define "opentelemetry-operator.controller.manager.service.cert.name" -}}
-opentelemetry-operator-controller-manager-service-cert
+{{- define "opentelemetry-operator.controller.manager.metrics.service.url" -}}
+http://opentelemetry-operator-controller-manager-metrics-service.{{ .Release.Namespace }}:8080/metrics
 {{- end -}}
 
 {{/*
-Generate certificates for opentelemetry-operator webhook
-
-Example usage:
-{{ $certManagerEnabled := index .Values "opentelemetry-operator" "admissionWebhooks" "certManager" "enabled" }}
-{{ $operatorEnabled := index .Values "opentelemetry-operator" "enabled" }}
-{{- if eq ( not $certManagerEnabled ) $operatorEnabled }}
-{{ ( include "opentelemetry-operator.non_certmanager" . ) }}
-{{- end }}
-
+Generate Instrumentation Custom Resource for specified namespaces.
 */}}
-{{- define "opentelemetry-operator.non_certmanager" -}}
-{{- $altNames := list ( printf "%s.%s" (include "opentelemetry-operator.webhook.service.name" .) .Release.Namespace ) ( printf "%s.%s.svc" (include "opentelemetry-operator.webhook.service.name" .) .Release.Namespace ) ( printf "%s.%s.svc.cluster.local" (include "opentelemetry-operator.webhook.service.name" .) .Release.Namespace ) -}}
-{{- $ca := genCA ( include "opentelemetry-operator.webhook.service.name" . ) 365 -}}
-{{- $cert := genSignedCert ( include "opentelemetry-operator.webhook.service.name" . ) nil $altNames 365 $ca -}}
-apiVersion: v1
-kind: Secret
-type: kubernetes.io/tls
-metadata:
-  name: {{ include "opentelemetry-operator.controller.manager.service.cert.name" . }}
-  labels:
-    {{- include "sumologic.labels.common" . | nindent 4 }}
-data:
-  tls.crt: {{ $cert.Cert | b64enc }}
-  tls.key: {{ $cert.Key | b64enc }}
+
+{{- define "opentelemetry-operator.instrumentation.custom.resource" -}}
+{{- $ctx := . -}}
+{{ $watchNamespace := index .Values "opentelemetry-operator" "manager" "env" }}
+{{- if eq ( get $watchNamespace "WATCH_NAMESPACE" ) "" }}
+{{ fail "It is mandatory to set value for \"opentelemetry-operator.manager.env.WATCH_NAMESPACE\" when opentelemetry-operator is enabled. Value is comma separated namespaces e.g. \"ns1\\,ns2\"" }}
+{{ else }}
+{{- range $ns := splitList "," ( index .Values "opentelemetry-operator" "manager" "env" "WATCH_NAMESPACE" ) }}
 ---
-apiVersion: admissionregistration.k8s.io/v1
-kind: MutatingWebhookConfiguration
+apiVersion: opentelemetry.io/v1alpha1
+kind: Instrumentation
 metadata:
-  annotations:
-    cert-manager.io/inject-ca-from: {{ printf "%s/%s" .Release.Namespace ( include "opentelemetry-operator.controller.manager.service.cert.name" . ) }}
+  namespace: {{ $ns }}
+  name: {{ template "sumologic.metadata.name.opentelemetry.operator.instrumentation" $ctx }}
```

Comment:
I would put it in separate file

Reply:
I've moved `Instrumentation` CR generation to separated file located in `confs/opentelemetry-operator`